### PR TITLE
Issue #106 - JSR 380 Bean Validation 2.0

### PR DESCRIPTION
**Pull request by gala-lightfoot**
Validate inbound TradeAgreement's:
- Side (B/S/SS)
- Qty (> 0)

Validate SettlementMissions:
- Direction (DEL/REC)
- Qty (> 0)

### CODE REVIEW

File: src/main/java/org/galatea/starter/entrypoint/RestExceptionHandler.java [hunk 1]
```diff
@@ -44,6 +46,16 @@
     return buildResponseEntity(error);
   }
 
+  @ExceptionHandler(ConstraintViolationException.class)
+  protected ResponseEntity<Object> handleConstraintViolation(
+      final ConstraintViolationException exception) {
+    log.debug("Invalid input data sent", exception);
+
+    String errorMessage = "Invalid input data.  Please consult the documentation.";
```
Comment (paulo-reichert) [inline src/main/java/org/galatea/starter/entrypoint/RestExceptionHandler.java]: Can we give the caller more info about what data failed validation? Like when filling a web form, would be useful to know what field fell foul of the rules.

File: src/test/java/org/galatea/starter/domain/SettlementMissionTest.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+import org.junit.Test;
+
+import java.util.Set;
+
+import javax.validation.ConstraintViolation;
+import javax.validation.Validation;
+import javax.validation.Validator;
+import javax.validation.ValidatorFactory;
+
+public class SettlementMissionTest {
```
Comment (paulo-reichert) [inline src/test/java/org/galatea/starter/domain/SettlementMissionTest.java:15]: For completeness, this should also test success scenarios.

File: src/main/java/org/galatea/starter/MethodValidationConfig.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+ */
+@Configuration
+@ComponentScan("org.galatea.starter")
+public class MethodValidationConfig {
+
+  /**
+   * Delegates the Bean Validation to a provider (in our case, the Hibernate Validator).
+   */
+  @Bean
+  public MethodValidationPostProcessor methodValidationPostProcessor() {
```
Comment (paulo-reichert) [inline src/main/java/org/galatea/starter/MethodValidationConfig.java:19]: Where do we use this?

Comment (gala-lightfoot) [reply src/main/java/org/galatea/starter/MethodValidationConfig.java:19]: We don't use it directly.  This is what causes the ConstraintViolationExceptions to be thrown when invalid data is detected (without out, the violations silently slip through the system).

**Review by paulo-reichert — APPROVED** [review]

**Comment by j-barber** [comment]
Should buySell and direction be enums instead of Strings?

**Pull request merged**